plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts

```diff
@@ -705,105 +710,151 @@ export class ArgoService implements ArgoServiceApi {
-    let isAppExist: boolean = true;
-    try {
-      const deleteAppResp = await this.deleteApp({
+    if (terminateOperation) {
+      const terminateOperationResp = await this.terminateArgoAppOperation({
         baseUrl: matchedArgoInstance.url,
-        argoApplicationName: argoAppName,
+        argoAppName: argoAppName,
         argoToken: token,
       });
-      if (deleteAppResp === false) {
-        countinueToDeleteProject = false;
-        argoDeleteAppResp.status = 'failed';
-        argoDeleteAppResp.message = 'error with deleteing argo app';
-      }
-    } catch (e: any) {
-      if (typeof e.message === 'string') {
-        isAppExist = false;
-        countinueToDeleteProject = true;
-        argoDeleteAppResp.status = 'failed';
-        argoDeleteAppResp.message = e.message;
-      }
-      const message =
-        e.message || `Failed to delete your app:, ${argoAppName}.`;
-      this.logger.info(message);
-    }
-
-    let isAppPendingDelete: boolean = false;
-    if (isAppExist) {
-      for (
-        let attempts = 0;
-        attempts < (this.config.getOptionalNumber('argocd.waitCycles') || 5);
-        attempts++
+      if (
+        terminateOperationResp.statusCode !== (404 || 200) &&
+        'message' in terminateOperationResp
       ) {
-        try {
-          const argoApp = await this.getArgoAppData(
-            matchedArgoInstance.url,
-            matchedArgoInstance.name,
-            token,
-            { name: argoAppName },
-          );
-
-          isAppPendingDelete = 'metadata' in argoApp;
-          if (!isAppPendingDelete) {
-            argoDeleteAppResp.status = 'success';
-            argoDeleteAppResp.message = 'application is deleted successfully';
-            break;
-          }
-
-          await timer(5000);
-        } catch (e: any) {
-          const message =
-            e.message || `Failed to get argo app data for:, ${argoAppName}.`;
-          this.logger.info(message);
-          if (
-            attempts ===
-            (this.config.getOptionalNumber('argocd.waitCycles') || 5) - 1
-          ) {
-            argoDeleteAppResp.status = 'failed';
-            argoDeleteAppResp.message = 'error getting argo app data';
-          }
-          continue;
-        }
+        terminateOperationDetails = {
+          status: 'failed',
+          argoResponse: terminateOperationResp,
+          message: 'failed to terminate operation',
+        };
+      } else if (
+        terminateOperationResp.statusCode === 404 &&
+        'message' in terminateOperationResp
+      ) {
+        terminateOperationDetails = {
+          status: 'failed',
+          argoResponse: terminateOperationResp,
+          message: 'application not found',
+        };
+      } else if (terminateOperationResp.statusCode === 200) {
+        terminateOperationDetails = {
+          status: 'success',
+          argoResponse: terminateOperationResp,
+          message: 'application is current operation terminated',
```

Comment:
This wording could use few tweaks. Maybe `application ${argoAppName} terminated` or something similar?